### PR TITLE
(#1435) Added support for "events" in Hooks.create()

### PR DESCRIPTION
**Pull request by llorllale**
This PR is for #1435:
* Added extra parameter to `Hooks.create()`
* Deleted puzzle for #1309 because we didn't notice that issue was open (it's not in scope)
* Left puzzle to add tests

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#1437](https://github.com/jcabi/jcabi-github/issues/1437) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C9R53K5JA/2632132019); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by 0crat** [comment]
Job `gh:jcabi/jcabi-github#1437` already assigned to @paulodamaso, can't assign to @paulodamaso

File: src/main/java/com/jcabi/github/RtHooks.java [hunk 1]
```diff
@@ -118,20 +119,27 @@ public Hook get(final int number) {
...
         final boolean active
     ) throws IOException {
-        final JsonObjectBuilder builder = Json.createObjectBuilder();
+        final JsonObjectBuilder configs = Json.createObjectBuilder();
         for (final Map.Entry<String, String> entr : config.entrySet()) {
-            builder.add(entr.getKey(), entr.getValue());
+            configs.add(entr.getKey(), entr.getValue());
+        }
+        final JsonArrayBuilder evnts = Json.createArrayBuilder();
+        for (final Event event : events) {
```
Comment (paulodamaso) [inline src/main/java/com/jcabi/github/RtHooks.java]: @llorllale We could rewrite this `for` using `events.forEach()`, WDYT?

File: src/main/java/com/jcabi/github/mock/MkHooks.java [hunk 2]
```diff
@@ -137,7 +139,11 @@ public Hook create(
                 .add("id").set(String.valueOf(number)).up()
                 .add("name").set(name).up()
                 .add("active").set(Boolean.toString(active)).up()
-                .add("events").up()
+                .add("events");
+            for (final Event event : events) {
```
Comment (paulodamaso) [inline src/main/java/com/jcabi/github/mock/MkHooks.java]: @llorllale Same here, we could rewrite this `for` using `events.forEach()`.

File: src/main/java/com/jcabi/github/RtHooks.java (hunk 1 above)
Comment (llorllale) [reply src/main/java/com/jcabi/github/RtHooks.java]: @paulodamaso I would be happy to use Java 8 features, but somewhere in the build (maybe parents?) the source level is being set to 1.6.

File: src/main/java/com/jcabi/github/mock/MkHooks.java (hunk 2 above)
Comment (llorllale) [reply src/main/java/com/jcabi/github/mock/MkHooks.java]: @paulodamaso see previous comment

**Comment by llorllale** [comment]
@paulodamaso I opened #1438

**Comment by paulodamaso** [comment]
@llorllale Right, just please add a `@todo` to the code saying that these two iterations should be rewritten using Java 8 features after https://github.com/jcabi/jcabi-github/issues/1438 implementation so this task can be solved later

**Comment by llorllale** [comment]
@paulodamaso I think that `@todo` you're asking for can be added if/when #1438 is included in scope

**Comment by paulodamaso** [comment]
@llorllale I think that it'll trouble this ticket then: this way I must have to wait for the todo addition to approve this review. I've already seen in several other projects and issues this kind of `@todo` chaining ("resolve this one after some other one is solved") so I think that wouldn't do any harm here.

**Comment by llorllale** [comment]
@paulodamaso you don't have to wait for #1438 because the decision to upgrade to 1.8 hasn't been made yet.

Only once that decision has been made *then* it'll make sense to add `@todo`s to refactor code.

Make sense?

**Comment by paulodamaso** [comment]
@llorllale Sort of, but this way we "lost" this `@todo`, don't you agree? So please add it toto https://github.com/jcabi/jcabi-github/issues/1438 then, we should document it someway.

**Comment by llorllale** [comment]
@paulodamaso 

> don't you agree?

No, I don't. :(

**Comment by paulodamaso** [comment]
@llorllale So, who will create this issue if/when #1438 is included in scope?

**Comment by llorllale** [comment]
@paulodamaso whoever works on #1438 (or any child issues), if it's accepted

**Comment by paulodamaso** [comment]
@llorllale How this person will know about this problem?

**Comment by llorllale** [comment]
@amihaiemil please help :)

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by amihaiemil** [comment]
@paulodamaso @llorllale Looks ok to me. But do you want to update to Java 8 only to use ``.forEach()``  mehtod instead of the classic ``foreach (:)`` statement which relise on the iterator?

**Comment by llorllale** [comment]
@amihaiemil there's also the redundant generic type declaration on the RHS of expressions

**Comment by llorllale** [comment]
@amihaiemil ping

**Comment by amihaiemil** [comment]
@rultor merge again?

**Comment by 0crat** [comment]
Code review was too long (16 days), architects (@amihaiemil) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**